src/js/background.js

```diff
@@ -732,17 +698,18 @@ Badger.prototype = {
 /**************************** Listeners ****************************/
 
 function startBackgroundListeners() {
-  chrome.webRequest.onBeforeRequest.addListener(function(details) {
-    if (details.tabId != -1){
-      badger.updateCount(details);
-    }
-  }, {urls: ["http://*/*", "https://*/*"]}, []);
-
-
-  // Update icon if a tab changes location
+  // Update icon if a tab changes location. If new location is a background
+  // tab or Privacy Badger is disabled on the new origin, hide the badge
+  // counter. Else refresh the badge counter to '0'.
   chrome.tabs.onUpdated.addListener(function(tabId, changeInfo, tab) {
     if(changeInfo.status == "loading") {
       badger.refreshIconAndContextMenu(tab);
+      if ((changeInfo.url && changeInfo.url.indexOf("http") !== 0)
+        || badger.isPrivacyBadgerDisabled(webrequest.getHostForTab(tabId))) {
```

Comment:
I should have documented this better. There was a case where `changeInfo.status` was `"loading"` but the URL was null. I can't remember exactly what was causing it - maybe when opening a background page from a new tab? Anyway, I recall it was throwing errors in the console before I added this workaround in a new commit.